index.js

```diff
@@ -380,24 +385,26 @@ function handleRequests() {
   /** everything was applied correctly; reset the global variable. */
   responseMiddlewareHasBeenApplied = false;
 
-  /** middleware to handle REQUESTS */
-  app.use((req, res, next) => {
-    try {
-      const methodAndPathKey = getMethod(req);
-      if (methodAndPathKey && methodAndPathKey.method && methodAndPathKey.pathKey) {
-        const method = methodAndPathKey.method;
-        updateSchemesAndHost(req);
-        processors.processPath(req, method, methodAndPathKey.pathKey);
-        processors.processHeaders(req, method, spec);
-        processors.processBody(req, method);
-        processors.processQuery(req, method);
+  if (!ignoredNodeEnvironments.includes(process.env.NODE_ENV)) {
+    /** middleware to handle REQUESTS */
+    app.use((req, res, next) => {
+      try {
+        const methodAndPathKey = getMethod(req);
+        if (methodAndPathKey && methodAndPathKey.method && methodAndPathKey.pathKey) {
+          const method = methodAndPathKey.method;
+          updateSchemesAndHost(req);
+          processors.processPath(req, method, methodAndPathKey.pathKey);
+          processors.processHeaders(req, method, spec);
+          processors.processBody(req, method);
+          processors.processQuery(req, method);
+        }
+      } catch (e) {
+        /** TODO - shouldn't we do something here? */
+      } finally {
+        next();
       }
-    } catch (e) {
-      /** TODO - shouldn't we do something here? */
-    } finally {
-      next();
-    }
-  });
+    });
+  }
 
   /** forward options to `serveApiDocs`: */
   serveApiDocs();
```

Comment:
why that line is called in any NODE_ENV?

Reply:
I think we need to difference between:

1. Intercepting with the middlewares. 
2. Serving the docs/spec

Users could want to use (2) without intercepting. That's why I suggested 2 different boolean args.

Reply:
Not serving anymore now, but I think we should review that.